components/cdc/src/endpoint.rs

```diff
@@ -1437,6 +1493,53 @@ impl Initializer {
             error!("cdc schedule cdc task failed"; "error" => ?e);
         }
     }
+
+    fn ts_filter_is_helpful<S: Snapshot>(&self, snap: &S) -> bool {
+        if self.ts_filter_ratio < f64::EPSILON {
+            return false;
+        }
+
+        let start_key = data_key(snap.lower_bound().unwrap_or_default());
+        let end_key = data_end_key(snap.upper_bound().unwrap_or_default());
+        let range = Range::new(&start_key, &end_key);
+        let collection = match self.engine.table_properties_collection(CF_WRITE, &[range]) {
+            Ok(collection) => collection,
+            Err(_) => return false,
+        };
+
+        let hint_min_ts = self.checkpoint_ts.into_inner();
+        let (mut total_count, mut filtered_count, mut tables) = (0, 0, 0);
+        collection.iter_user_collected_properties(|prop| {
+            tables += 1;
+            if let Some((_, keys)) = prop.approximate_size_and_keys(&start_key, &end_key) {
```

Comment:
Here @tabokie . I think comparing with the scan itself, this price is ok.

Reply:
Got it